native/membrane_videocompositor/src/elixir_bridge/elixir_structs.rs

```diff
@@ -0,0 +1,138 @@
+#[module = "Membrane.VideoCompositor.RustStructs.BaseVideoPlacement"]
+pub struct ElixirBaseVideoPlacement {
+    pub position: (i32, i32),
+    pub size: (u32, u32),
+    pub z_value: f32,
+}
+
+impl Into<VideoPlacement> for ElixirBaseVideoPlacement {
+    fn into(self) -> VideoPlacement {
+        compositor::VideoPlacement {
+            position: Vec2d {
+                x: self.position.0,
+                y: self.position.1,
+            },
+            size: Vec2d {
+                x: self.size.0,
+                y: self.size.1,
+            },
+            z: convert_z(self.z_value),
+        }
+    }
+}
+
+/// Describes all transformations applied to video
+#[derive(Debug, rustler::NifStruct)]
+#[module = "Membrane.VideoCompositor.VideoTransformations"]
+pub struct ElixirVideoTransformations {
+    pub texture_transformations: Vec<ElixirTextureTransformations>,
+}
+
+impl Into<Vec<Box<dyn TextureTransformation>>> for ElixirVideoTransformations {
+    fn into(self) -> Vec<Box<dyn TextureTransformation>> {
+        let mut texture_transformations = Vec::new();
+
+        for texture_transformation in self.texture_transformations.into_iter() {
+            texture_transformations.push(texture_transformation.into());
+        }
+
+        texture_transformations
+    }
+}
+
+/// Wraps video transformations parameters (wrapped in structs) into enum.
+/// Allows passing to rust elixir texture transformation type,
+/// which is algebraic sum type of all structs describing single
+/// texture transformation.
+/// As a developer adding new texture transformation, you need just to add
+/// new enum value and implement new match arm converting elixir structs to
+/// rust structs used in shader.
+#[derive(Debug, NifUntaggedEnum, Clone, Copy)]
+pub enum ElixirTextureTransformations {
+    CornersRounding(ElixirCornersRounding),
+    Cropping(ElixirCropping),
+}
+
+impl Into<Box<dyn TextureTransformation>> for ElixirTextureTransformations {
+    fn into(self) -> Box<dyn TextureTransformation> {
+        match self {
+            ElixirTextureTransformations::CornersRounding(elixir_corners_rounding) => {
+                elixir_corners_rounding.into()
+            }
+            ElixirTextureTransformations::Cropping(elixir_cropping) => elixir_cropping.into(),
+        }
+    }
+}
+
+/// Elixir struct wrapping parameters describing corner rounding texture transformation
+#[derive(Debug, rustler::NifStruct, Clone, Copy)]
+#[module = "Membrane.VideoCompositor.VideoTransformations.TextureTransformations.CornersRounding"]
+pub struct ElixirCornersRounding {
+    pub border_radius: u32,
+}
+
+impl Into<Box<dyn TextureTransformation>> for ElixirCornersRounding {
+    fn into(self) -> Box<dyn TextureTransformation> {
+        Box::new(CornersRounding {
+            border_radius: self.border_radius as f32,
+            video_width: 0.0,  // will be updated in compositor
+            video_height: 0.0, // will be updated in compositor
```

Comment:
If what I wrote in the previous comment works, we should be able to get rid of this awfulness

Reply:
up

Reply:
I deleted this